homeassistant/components/media_player/denon_http.py

```diff
@@ -0,0 +1,678 @@
+             'Media Player': 'MPLAY', 'CD': 'CD', 'AUX2': 'AUX2',
+             'HD Radio': 'HDRADIO', 'AUX': 'AUX1', 'Bluetooth': 'BT',
+             'CBL/SAT': 'SAT/CBL', 'iPod/USB': 'USB/IPOD', 'Phono': 'PHONO',
+             'Blu-ray': 'BD', 'Media Server': 'SERVER'}
+
+PLAYING_SOURCES = ("Online Music", "Media Server", "iPod/USB", "Bluetooth",
+                   "Internet Radio", "Tuner", "HD Radio")
+NETAUDIO_SOURCES = ("Online Music", "Media Server", "iPod/USB", "Bluetooth",
+                    "Internet Radio")
+
+STATUS_URL = "/goform/formMainZone_MainZoneXmlStatus.xml"
+MAINZONE_URL = "/goform/formMainZone_MainZoneXml.xml"
+NETAUDIOSTATUS_URL = "/goform/formNetAudio_StatusXml.xml"
+TUNERSTATUS_URL = "/goform/formTuner_TunerXml.xml"
+HDTUNERSTATUS_URL = "/goform/formTuner_HdXml.xml"
+COMMAND_SEL_SRC_URL = "/goform/formiPhoneAppDirect.xml?SI"
+COMMAND_POWER_ON_URL = "/goform/formiPhoneAppPower.xml?1+PowerOn"
+COMMAND_POWER_STANDBY_URL = "/goform/formiPhoneAppPower.xml?1+PowerStandby"
+COMMAND_VOLUME_UP_URL = "/goform/formiPhoneAppDirect.xml?MVUP"
+COMMAND_VOLUME_DOWN_URL = "/goform/formiPhoneAppDirect.xml?MVDOWN"
+COMMAND_MUTE_ON_URL = "/goform/formiPhoneAppMute.xml?1+MuteOn"
+COMMAND_MUTE_OFF_URL = "/goform/formiPhoneAppMute.xml?1+MuteOff"
+COMMAND_NETAUDIO_POST_URL = "/NetAudio/index.put.asp"
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Denon platform."""
+    denon = DenonDevice(config.get(CONF_NAME), config.get(CONF_HOST))
+
+    if denon.update():
+        add_devices([denon])
+        _LOGGER.info("Denon receiver at host %s initialized",
+                     config.get(CONF_HOST))
+        return True
+    else:
+        _LOGGER.error("Denon receiver at host %s could not be initialized",
+                      config.get(CONF_HOST))
+        return False
+
+
+class DenonDevice(MediaPlayerDevice):
+    """Representation of a Denon Media Player Device."""
+
+    def __init__(self, name, host):
+        """Initialize the device."""
+        self._name = name
+        self._host = host
+        self._muted = False
+        self._volume = "--"
+        self._current_source = None
+        self._source_list = {}
+        self._state = STATE_UNKNOWN
+        self._power = STATE_UNKNOWN
+        self._modelid = None
+        self._salesarea = None
+        self._media_image_url = (
+            "http://{host}/img/album%20art_S.png".format(host=host))
+        self._media_title = None
+        self._media_artist = None
+        self._media_album_name = None
+
+    def update(self):
+        """Get the latest status information from device."""
+        # Get status XML from Denon receiver via http
+        try:
+            root = self.get_status_xml(self._host, MAINZONE_URL)
+        except ConnectionError:
+            self._state = STATE_OFF
+            return False
+
+        # Get the relevant tags from the XML structure
+        for child in root:
```

Comment:
Please extract the implementation on how to communicate with this device into a lib that gets required in Home Assistant.